src/swish/supervisor.ss

```diff
@@ -352,23 +354,27 @@
          [brutal-kill
           (kill pid 'kill)
           (receive
-           [#(DOWN ,_ ,@pid ,reason) reason])]
+           [`(DOWN ,_ ,@pid ,reason ,err)
+            (values reason err)])]
          [,timeout
           (kill pid 'shutdown)
           (receive (after timeout
                      (kill pid 'kill)
                      (receive
-                      [#(DOWN ,_ ,@pid ,reason) reason]))
-            [#(DOWN ,_ ,@pid ,reason) reason])])]
-      [#(error ,reason) reason]))
+                      [`(DOWN ,_ ,@pid ,reason ,err)
+                       (values reason err)]))
+            [`(DOWN ,_ ,@pid ,reason ,err)
+             (values reason err)])])]
+      [#(error ,reason)
+       (values reason #f)]))
 
   (define (monitor-child pid)
     (monitor pid)
     (unlink pid)
     (receive (after 0 'ok)
-      [#(EXIT ,@pid ,reason)
+      [`(EXIT ,@pid ,reason ,err)
```

Comment:
`err` does not appear to be used. Same for `,err` below.

Reply:
Good point. I've updated it to use the value and added some tests.